lib/src/client/connection.dart

```diff
@@ -43,24 +95,38 @@ class Connection {
   }
 
   /// Replaces the current event handlers.
-  void listen(void Function(List<int> data) onData, void Function() onDone) {
+  void listen(void Function(List<int> data) onData,
```

Comment:
Maybe all these parameters could be optional now, I don't like the `(nul, null, ..)`  construction

Reply:
I somewhat agree, we but only using this internally and in tests...

We could also argue that it shouldn't be possible to create more than one client with a `Connection`, right know you can create another client object and it'll just sort of hijack the data...

Reply:
Yes, don't worry too much about it.

I usually write a lot of comments, but I can accept your pull request and make the changes after.